From cfe735cd160984452bfe4ba7b49647c6be101364 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Oliver=20St=C3=A4bler?= <oliver.staebler@bytesatwork.ch>
Date: Mon, 6 Nov 2017 14:30:30 +0100
Subject: [PATCH] ARM: dts: byteboard: Change dts to dtsi

Change the dts into a dtsi so we can include the dtsi from other dts
---
 arch/arm/boot/dts/byteboard.dts  | 123 +----------------------------------
 arch/arm/boot/dts/byteboard.dtsi | 134 +++++++++++++++++++++++++++++++++++++++
 2 files changed, 136 insertions(+), 121 deletions(-)
 create mode 100644 arch/arm/boot/dts/byteboard.dtsi

diff --git a/arch/arm/boot/dts/byteboard.dts b/arch/arm/boot/dts/byteboard.dts
index 881e11e1e8b3..afb0942cddb2 100644
--- a/arch/arm/boot/dts/byteboard.dts
+++ b/arch/arm/boot/dts/byteboard.dts
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016 bytes at work AG - http://www.bytesatwork.ch
+ * Copyright (C) 2017 bytes at work AG - http://www.bytesatwork.ch
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as  published by the
@@ -14,123 +14,4 @@
 
 /dts-v1/;
 
-#include "byteengine-m3.dtsi"
-
-/ {
-	model = "byteBoard";
-	compatible = "batw,byteboard", "ti,am4372", "ti,am43";
-};
-
-&am43xx_pinmux {
-	uart0_pins_default: uart0_pins_default {
-		pinctrl-single,pins = <
-			PIN_UART0_RXD (PIN_INPUT_PULLUP | SLEWCTRL_FAST | DS0_PULL_UP_DOWN_EN | MUX_MODE0)		/* uart0_rxd.uart0_rxd */
-			PIN_UART0_TXD (PIN_INPUT | PULL_DISABLE | SLEWCTRL_FAST | DS0_PULL_UP_DOWN_EN | MUX_MODE0)	/* uart0_txd.uart0_txd */
-		>;
-	};
-
-	usb1_pins_default: usb1_pins_default {
-		pinctrl-single,pins = <
-			PIN_USB0_DRVVBUS (PIN_OUTPUT | MUX_MODE0)	/* usb0_drvvbus.usb0_drvvbus */
-		>;
-	};
-
-	cpsw_default: cpsw_default {
-		pinctrl-single,pins = <
-			PIN_MII1_TXEN (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txen.rgmii1_txen */
-			PIN_MII1_RXDV (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxdv.rgmii1_rxctl */
-			PIN_MII1_TXD3 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd1.rgmii1_txd3 */
-			PIN_MII1_TXD2 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd0.rgmii1_txd2 */
-			PIN_MII1_TXD1 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd1.rgmii1_txd1 */
-			PIN_MII1_TXD0 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd0.rgmii1_txd0 */
-			PIN_MII1_TXCLK (PIN_OUTPUT_PULLDOWN | MUX_MODE2)	/* mii1_txclk.rmii1_tclk */
-			PIN_MII1_RXCLK (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxclk.rmii1_rclk */
-			PIN_MII1_RXD3 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd1.rgmii1_rxd3 */
-			PIN_MII1_RXD2 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd0.rgmii1_rxd2 */
-			PIN_MII1_RXD1 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd1.rgmii1_rxd1 */
-			PIN_MII1_RXD0 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd0.rgmii1_rxd0 */
-			PIN_MII1_COL (PIN_OUTPUT_PULLDOWN | MUX_MODE7)		/* mii1_col.gpio3_0 */
-		>;
-	};
-
-	davinci_mdio_default: davinci_mdio_default {
-		pinctrl-single,pins = <
-			PIN_MDIO_DATA (PIN_INPUT_PULLUP | SLEWCTRL_FAST | MUX_MODE0)	/* mdio_data.mdio_data */
-			PIN_MDIO_CLK (PIN_OUTPUT_PULLUP | MUX_MODE0)			/* mdio_clk.mdio_clk */
-		>;
-	};
-
-	mmc1_pins: pinmux_mmc1_pins {
-		pinctrl-single,pins = <
-			PIN_MMC0_CLK (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_clk.mmc0_clk */
-			PIN_MMC0_CMD (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_cmd.mmc0_cmd */
-			PIN_MMC0_DAT3 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat0.mmc0_dat0 */
-			PIN_MMC0_DAT2 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat1.mmc0_dat1 */
-			PIN_MMC0_DAT1 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat2.mmc0_dat2 */
-			PIN_MMC0_DAT0 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat3.mmc0_dat3 */
-			PIN_SPI0_CS1 (PIN_INPUT | MUX_MODE7)		/* spi0_cs1.gpio0_6 */
-		>;
-	};
-};
-
-/* console */
-&uart0 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_pins_default>;
-};
-
-/* usb */
-&usb2_phy1 {
-	status = "okay";
-};
-
-&usb1 {
-	dr_mode = "host";
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&usb1_pins_default>;
-};
-
-/* ethernet */
-&mac {
-	slaves = <1>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&cpsw_default>;
-	status = "okay";
-};
-
-&davinci_mdio {
-	pinctrl-names = "default";
-	pinctrl-0 = <&davinci_mdio_default>;
-	status = "okay";
-};
-
-&cpsw_emac0 {
-	phy_id = <&davinci_mdio>, <0>;
-	phy-mode = "rgmii";
-};
-
-/* sd card */
-&mmc1 {
-	status = "okay";
-	vmmc-supply = <&vdd_3v3d>;
-	bus-width = <4>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&mmc1_pins>;
-	cd-gpios = <&gpio0 6 GPIO_ACTIVE_HIGH>;
-};
-
-&gpio0 {
-	status = "okay";
-};
-
-&gpio3 {
-	status = "okay";
-
-	p0 {
-		gpio-hog;
-		gpios = <0 GPIO_ACTIVE_HIGH>;
-		output-high;
-	};
-};
+#include "byteboard.dtsi"
diff --git a/arch/arm/boot/dts/byteboard.dtsi b/arch/arm/boot/dts/byteboard.dtsi
new file mode 100644
index 000000000000..652180b7cdf9
--- /dev/null
+++ b/arch/arm/boot/dts/byteboard.dtsi
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017 bytes at work AG - http://www.bytesatwork.ch
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as  published by the
+ * Free Software Foundation version 2.
+ *
+ * This program is distributed "as is" WITHOUT ANY WARRANTY of any kind,
+ * whether express or implied; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ */
+
+#include "byteengine-m3.dtsi"
+
+/ {
+	model = "byteBoard";
+	compatible = "batw,byteboard", "ti,am4372", "ti,am43";
+};
+
+&am43xx_pinmux {
+	uart0_pins_default: uart0_pins_default {
+		pinctrl-single,pins = <
+			PIN_UART0_RXD (PIN_INPUT_PULLUP | SLEWCTRL_FAST | DS0_PULL_UP_DOWN_EN | MUX_MODE0)		/* uart0_rxd.uart0_rxd */
+			PIN_UART0_TXD (PIN_INPUT | PULL_DISABLE | SLEWCTRL_FAST | DS0_PULL_UP_DOWN_EN | MUX_MODE0)	/* uart0_txd.uart0_txd */
+		>;
+	};
+
+	usb1_pins_default: usb1_pins_default {
+		pinctrl-single,pins = <
+			PIN_USB0_DRVVBUS (PIN_OUTPUT | MUX_MODE0)	/* usb0_drvvbus.usb0_drvvbus */
+		>;
+	};
+
+	cpsw_default: cpsw_default {
+		pinctrl-single,pins = <
+			PIN_MII1_TXEN (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txen.rgmii1_txen */
+			PIN_MII1_RXDV (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxdv.rgmii1_rxctl */
+			PIN_MII1_TXD3 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd1.rgmii1_txd3 */
+			PIN_MII1_TXD2 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd0.rgmii1_txd2 */
+			PIN_MII1_TXD1 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd1.rgmii1_txd1 */
+			PIN_MII1_TXD0 (PIN_OUTPUT_PULLDOWN | MUX_MODE2)		/* mii1_txd0.rgmii1_txd0 */
+			PIN_MII1_TXCLK (PIN_OUTPUT_PULLDOWN | MUX_MODE2)	/* mii1_txclk.rmii1_tclk */
+			PIN_MII1_RXCLK (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxclk.rmii1_rclk */
+			PIN_MII1_RXD3 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd1.rgmii1_rxd3 */
+			PIN_MII1_RXD2 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd0.rgmii1_rxd2 */
+			PIN_MII1_RXD1 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd1.rgmii1_rxd1 */
+			PIN_MII1_RXD0 (PIN_INPUT_PULLDOWN | MUX_MODE2)		/* mii1_rxd0.rgmii1_rxd0 */
+			PIN_MII1_COL (PIN_OUTPUT_PULLDOWN | MUX_MODE7)		/* mii1_col.gpio3_0 */
+		>;
+	};
+
+	davinci_mdio_default: davinci_mdio_default {
+		pinctrl-single,pins = <
+			PIN_MDIO_DATA (PIN_INPUT_PULLUP | SLEWCTRL_FAST | MUX_MODE0)	/* mdio_data.mdio_data */
+			PIN_MDIO_CLK (PIN_OUTPUT_PULLUP | MUX_MODE0)			/* mdio_clk.mdio_clk */
+		>;
+	};
+
+	mmc1_pins: pinmux_mmc1_pins {
+		pinctrl-single,pins = <
+			PIN_MMC0_CLK (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_clk.mmc0_clk */
+			PIN_MMC0_CMD (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_cmd.mmc0_cmd */
+			PIN_MMC0_DAT3 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat0.mmc0_dat0 */
+			PIN_MMC0_DAT2 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat1.mmc0_dat1 */
+			PIN_MMC0_DAT1 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat2.mmc0_dat2 */
+			PIN_MMC0_DAT0 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat3.mmc0_dat3 */
+			PIN_SPI0_CS1 (PIN_INPUT | MUX_MODE7)		/* spi0_cs1.gpio0_6 */
+		>;
+	};
+};
+
+/* console */
+&uart0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_pins_default>;
+};
+
+/* usb */
+&usb2_phy1 {
+	status = "okay";
+};
+
+&usb1 {
+	dr_mode = "host";
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&usb1_pins_default>;
+};
+
+/* ethernet */
+&mac {
+	slaves = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cpsw_default>;
+	status = "okay";
+};
+
+&davinci_mdio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&davinci_mdio_default>;
+	status = "okay";
+};
+
+&cpsw_emac0 {
+	phy_id = <&davinci_mdio>, <0>;
+	phy-mode = "rgmii";
+};
+
+/* sd card */
+&mmc1 {
+	status = "okay";
+	vmmc-supply = <&vdd_3v3d>;
+	bus-width = <4>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc1_pins>;
+	cd-gpios = <&gpio0 6 GPIO_ACTIVE_HIGH>;
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+
+	p0 {
+		gpio-hog;
+		gpios = <0 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
-- 
2.11.0

